applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -0,0 +1,338 @@
+            from KratosMultiphysics.mpi import mpi
+            import KratosMultiphysics.MetisApplication
+            import KratosMultiphysics.TrilinosApplication
+            import KratosMultiphysics.MappingApplication #TODO: Import always once we use the serial version of the mapper
+            self.is_printing_rank = (mpi.rank == 0)
+        else:
+            self.is_printing_rank = True
+
+        # Deprecation warnings
+        # This makes possible the FSI solver derivation from the core base python_solver.py
+        if not self.project_parameters.Has("echo_level"):
+            Kratos.Logger.PrintInfo("FSIAnalysis", "Using the old way to pass the echo_level, this will be removed!")
+            self.project_parameters.AddEmptyValue("echo_level")
+            self.project_parameters["echo_level"].SetInt(self.echo_level)
+
+        # Add solver variables (note that the solver is created in the first _GetSolver() call)
+        self._GetSolver().AddVariables()
+
+    def Initialize(self):
+        '''
+        Construct and initialize all classes and tools used in the simulation loop.
+        '''
+        self._SetUpRestart()
+
+        if self.load_restart:
+            fluid_restart_utility = self._GetFluidRestartUtility()
+            structure_restart_utility = self._GetStructureRestartUtility()
+            fluid_restart_utility.LoadRestart()
+            structure_restart_utility.LoadRestart()
+        else:
+            self._GetSolver().ImportModelPart()
+            self._GetSolver().PrepareModelPart()
+            self._GetSolver().AddDofs()
+            self.fluid_main_model_part = self.model.GetModelPart(
+                self.project_parameters["fluid_solver_settings"]["solver_settings"]["model_part_name"].GetString())
+            self.structure_main_model_part = self.model.GetModelPart(
+                self.project_parameters["structure_solver_settings"]["solver_settings"]["model_part_name"].GetString())
+
+        # This should let eventual derived stages modify the model after reading.
+        self.ModifyInitialProperties()
+        self.ModifyInitialGeometry()
+
+        self._SetUpListOfProcesses()
+        self._SetUpAnalysis()
+
+        for process in self._GetListOfProcesses():
+            process.ExecuteBeforeSolutionLoop()
+
+    def InitializeSolutionStep(self):
+
+        # Since no substepping is implemented yet, structure and fluid step/time must match
+        step_fluid = self.fluid_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        step_structure = self.structure_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        if step_fluid != step_structure:
+            err_msg =  'Fluid step: '+ str(step_fluid) + '\n'
+            err_msg += 'Structure step: '+ str(step_structure) + '\n'
+            err_msg += 'No substepping has been implemented yet. Fluid and structure step must match.'
+            raise Exception(err_msg)
+
+        if self.is_printing_rank:
```

Comment:
Instead of copy pasting lines 99-105 I would call the baseclass-InitializeSolutionStep

Reply:
It is not possible, since the baseclass-InitializeSolutionStep calls the GetComputingModelPart() of the solver. Since in the partitioned FSI solver there are two "computing model parts" (fluid and structure) this method is not implemented.

Reply:
I see
I also don't like the current solution, we should come up with a better one for this @RiccardoRossi @jcotela 
=> I think we only need the time/step from the solver, which could be implemented in a separate function
I will think abt it